### DISCUSSION TITLE
Using YOLOv8 to measure the size of objects with a smartphone camera

### DISCUSSION THREAD

**Discussion by Portan21** [ultralytics/ultralytics#10361] [Q&A]
Hello, can I use YOLOv8 to measure the size of an object with a phone camera?
e.g: Measuring the objects' size in real time while camera is moving
e.g: Taking pictures of objects in different view and measure it's size accurately
Thank you in advance!

**Comment by pderrenger** [comment]
Hello! 👋

Yes, you can use YOLOv8 for object detection to identify objects within images or real-time video streams from a smartphone camera. However, measuring the actual size of detected objects involves additional steps beyond detection, as YOLOv8 will primarily provide bounding box coordinates of detected objects.

For measuring object size, you might consider combining YOLOv8's object detection capabilities with depth sensing or known scale references within the image if your device supports it. Calculating actual size accurately requires understanding the camera's intrinsics and the distance between the camera and the object, which aren't directly provided by YOLOv8.

In real-time scenarios or varying views, maintaining consistent size measurements could be challenging and might require custom calibration and algorithms tailored to your specific use case.

Here's a basic example of how you might start with YOLOv8 for object detection:

```python
from ultralytics import YOLO

# Load the YOLO model
model = YOLO('yolov8n.pt')  # Or your custom trained model

# Run detection
results = model('path/to/your/image.jpg')

# Process results
for det in results.pred[0]:
    xyxy = det[:4]  # Bounding box coordinates
    conf = det[4]   # Confidence
    cls = det[5]    # Detected class
```

For measuring sizes, you'd need to integrate additional logic based on your application requirements.

Best of luck with your project, and feel free to reach out if you have more questions! 🚀

**Reply by Portan21** [reply]
Okay, got it! Thank you very much!

**Reply by pderrenger** [reply]
You're welcome! If you have any more questions or need further assistance as you proceed, feel free to drop a message. Happy coding! 😊
